pkgs/servers/matrix-synapse/matrix-appservice-irc/default.nix

```
@@ -7,6 +7,27 @@ let
   };
   version = (lib.importJSON ./package.json).version;
   srcInfo = lib.importJSON ./src.json;
+
+  # TODO: package matrix-rust-sdk and use that instead of fetching & patching binaries
+  platformString =
+    if stdenv.hostPlatform.system == "x86_64-linux" then "linux-x64-gnu"
+    else if stdenv.hostPlatform.system == "aarch64-linux" then "linux-arm64-gnu"
+    else throw "matrix-sdk-crypto-nodejs: Architecture not supported";
+
+  matrix-sdk-crypto = let
+    version = "0.1.0-beta.1";
+    base = "https://github.com/matrix-org/matrix-rust-sdk/releases/download/matrix-sdk-crypto-nodejs-v${version}";
+    platformHash =
```

Maybe this is better?

```suggestion
    platformHash = {
      "x86_64-linux" = "sha256-a6FX+KhHooipIMsJ7Fl7gmUBt8WbTUgT6sXN4N3NXRk=";
      "aarch64-linux" = "sha256-jvr6gMTQ4aAk5x0iXpA28ADdaCgDpvOjmogd52Z6bIY=";
    };
  in
    fetchurl {
      url = "${base}/matrix-sdk-crypto.${platformString}.node";
      hash = platformHash.${stdenv.hostPlatform.system};
    };
```

---

Fair, in the end I was just copy/pasting stuff around to get it done.